example/flutter_example/lib/main.dart

```diff
@@ -22,7 +22,31 @@ class HomePage extends StatefulWidget {
 }
 
 class _HomePageState extends State<HomePage> {
-  Ros ros = Ros(url: 'ws://10.0.2.2:9090');
+  Ros ros;
+  Topic chatter;
+
+  @override
+  void initState() {
+    ros = Ros(url: 'ws://10.0.2.2:9090');
+    chatter = Topic(
+        ros: ros, name: '/chatter', type: "std_msgs/String", reconnectOnClose: true, queueLength: 10, queueSize: 10);
+    this.initConnection();
+    super.initState();
+  }
+
+  void initConnection() {
+    setState(() {
```

Comment:
`chatter.subscribe()` is asynchronous so I'm pretty sure the only reason this works right now is because the current implementation of `Topic.subscribe` sets its subscription stream before doing any async work. I'm not positive, and this works so I'm fine with how it currently is, but if you wanted to change it I'd suggest doing something either like this:
```
ros.connect();
chatter.subscribe().then(() => setState(() {}));
```
or this:
```
void initConnection() async {
  ros.connect();
  await chatter.subscribe();
  setState(() {});
}
```

Reply:
Second version is cleaner in my opinion. I'll do that.